controllers/vspherevm_controller_test.go

```diff
@@ -38,6 +38,25 @@ func TestReconcileNormal_WaitingForIPAddrAllocation(t *testing.T) {
 	s := model.Service.NewServer()
 	defer s.Close()
 
+	vsphereCluster := &infrav1.VSphereCluster{
```

Comment:
can we add a test to ensure we're picking up the right credentials (i.e. the ones in the `identityRef` for the VMs)

Reply:
added. The existing tests set the credentials through the controller manager context. the new test skips that and only sets it through the cluster